src/Bicep.Core/Configuration/RootConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Bicep.Core.Configuration
+{
+    public class RootConfiguration
+    {
+        private RootConfiguration(CloudConfiguration cloud, ModuleAliasesConfiguration moduleAliases, AnalyzersConfiguration analyzers, string resourceName)
```

Comment:
Mind adding a comment to explain what `resourceName` is in this context? Is that file URI / manifest resource name?

Reply:
Done.